src/EditorFeatures/Core/GoToDefinition/AbstractGoToSymbolService.cs

```diff
@@ -36,6 +36,7 @@ public async Task GetSymbolsAsync(GoToSymbolContext context)
             // This means we have to query for "third party navigation", from
             // XAML, etc. That call has to be done on the UI thread.
             await ThreadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, cancellationToken);
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
this makes me think we need a SafeSwitchToMainThreadAsync which codifies this pattern.

Reply:
I would probably want to call it something a bit more precise than "safe". Perhaps `SwitchToMainThreadIfNotCanceledAsync`.

But adding such an extension method would thwart the analyzers from understanding that you're on the UI thread afterward. We could add an AdditionalFiles feature if you go this route so that you can define that your extension method has similar behavior to the original method, so the analyzers can still work for you.

Reply:
> But adding such an extension method would thwart the analyzers from understanding that you're on the UI thread afterward.

Not really. The method would maintain the post-condition that not throwing implies the current thread is the main thread.

Reply:
@sharwell It's not that the method would behave significantly different -- it's that the analyzer wouldn't know it. 
Consider VSTHRD100 (which throws if calling STA objects without affirming you are on the main thread). Right now when you call the regular method, it can tell that STA calls are safe to make afterward. But if you use an extension method instead, the analyzer will misfire on STA calls after that line because it doesn't know that your new extension method switches to the main thread.